Removes Direct Accessing to Nova DB

muroi opened the pull request:
Masakari shouldn't access Nova DB to check instance's information in any case. For the first implementation, Masakari has to access Nova DB directly since it aims at Juno release, and Nova of the release has some constraints about evacuation of instances.  On the other hands, there is no constraint that Masakari hits to in the latest or supported version of Nova for the evacuation.

muroi committed:
Removes direct accesses to NovaDB

Accessing directly Nova DB is horrible since Masakari is not Nova,
so Masakari's codes have to only use Nova API to get instance's
state.

muroi committed:
Adds Some Unit Tests for Masakari-Controller

This patch adds some unit tests for masakari_worker.py
Usage:
 $ python tests/test_masakari_worker.py

muroi committed:
Adds a doc about Masakari's evacuation pattern.

The doc describes how Masakari reacts events monitored by monitoring
processes.

muroi committed:
Renames test file

sampathP commented:
@muroi  Thank you for fix. 
agree to merge..

toshikazu0314 commented on docs/evacuation_patterns.md:
```diff
@@ -0,0 +1,31 @@
+| Events | Monitored by | Previous instance's status | Rescue steps | Post instance's status |
+| :--- | :--- | :--- | :--- | :--- |
+| instance down | instancemonitor | active | nova.stop -> nova.start | active |
+| instance down | instancemonitor | stopped | nova.stop | stopped |
```
"nova.stop" should be "nova.reset" in case of an instance down of 'stopped' status instance.
Optionally, adding the status note of nova.reset like nova.reset(stop) would be beneficial to understand.

muroi replied:
Thanks for reviewing. I'll update the doc.

muroi committed:
Adds reset status in evacuation_patterns.md

To be easy to understand Masakari's behavior, adding new status reseted
by nova reset status.

toshikazu0314 commented:
Looks good.
